testsuite/pytests/sli2py_stimulating/test_step_current_generator.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+#
+# test_step_current_generator.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+"""
+Test basic properties of ``step_current_generator``.
+
+This set of tests checks parameter setting and compares actual simulation
+result against expectation for one application.
+"""
+
+import nest
+import numpy.testing as nptest
+import pandas as pd
+import pandas.testing as pdtest
+import pytest
+
+
+@pytest.fixture(autouse=True)
```

Comment:
Don't we have this globally somewhere?

Reply:
The global `safety_reset` in `conftest.py` only resets the kernel for each module. An local reset is needed in order to reset the kernel between tests within a module.